### PR TITLE
document wire.Struct first argument

### LINKED ISSUE #196: Document wire.Struct's first argument form
In particular, I am confused as to whether the first argument permits a struct value (`Foo{}`), or only a struct pointer (`new(Foo)`).

### PR DESCRIPTION
**Pull request by shantuo**
Fixes #196

### CODE REVIEW

File: wire.go [hunk 1]
```diff
@@ -146,10 +146,13 @@ func InterfaceValue(typ interface{}, x interface{}) ProvidedValue {
 // A StructProvider represents a named struct.
 type StructProvider struct{}
 
-// Struct specifies that the given struct type will be provided by filling in the fields
-// in the struct that have the names given. Each of the arguments must be a name
-// to the field they wish to reference. As a special case, if a single name "*"
-// is given, then all of the fields in the struct will be filled in.
+// Struct specifies that the given struct type will be provided by filling in the fields in the struct that have the names given.
+//
+// The first argument must be a pointer to the struct type, wire will analyze
```
Comment (zombiezen) [inline wire.go]:
```suggestion
// The first argument must be a pointer to the struct type. Wire will analyze
```

File: wire.go [hunk 2]
```diff
@@ -146,10 +146,13 @@ func InterfaceValue(typ interface{}, x interface{}) ProvidedValue {
...
+// and produce the struct type or the pointer to the struct type appropriatedly.
```
Comment (zombiezen) [inline wire.go]: For this sentence:

> For a struct type Foo, Wire will use field-filling to provide both Foo and *Foo.

File: wire.go [hunk 3]
```diff
@@ -146,10 +146,13 @@ func InterfaceValue(typ interface{}, x interface{}) ProvidedValue {
...
+// Each of the rest arguments must be a name to the field they wish to
```
Comment (zombiezen) [inline wire.go]: For this sentence:

> The remaining arguments are field names to fill in.

File: wire.go (hunk 1 above)
Comment (shantuo) [reply wire.go]: Done.

File: wire.go (hunk 2 above)
Comment (shantuo) [reply wire.go]: Done.

File: wire.go (hunk 3 above)
Comment (shantuo) [reply wire.go]: Done.

**Pull request merged**